demos/animation/anim.go

```diff
@@ -61,18 +61,18 @@ func (game *GameWorld) CreateEntity(point *engi.Point, spriteSheet *engi.Sprites
 }
 
 type ControlSystem struct {
-	*ecs.System
+	*ecs.BasicSystemWrapper
 }
 
-func (ControlSystem) Type() string {
-	return "ControlSystem"
-}
+func (*ControlSystem) Type() string { return "ControlSystem" }
+func (*ControlSystem) Pre()         {}
+func (*ControlSystem) Post()        {}
 
 func (c *ControlSystem) New(*ecs.World) {
-	c.System = ecs.NewSystem()
+	c.BasicSystemWrapper = ecs.NewBasicSystemWrapper(c)
 }
 
-func (c *ControlSystem) Update(entity *ecs.Entity, dt float32) {
+func (c *ControlSystem) EUpdate(entity *ecs.Entity, dt float32) {
```

Comment:
Yeah, I don't like this function name.

Also: What is the difference here? The function signatures look identical, I don't think people could tell just be looking.